### ISSUE TITLE
C++ can't link against user's installed version of TF

### ISSUE THREAD

**Issue by tommadams** [tensorflow/minigo#836]
**tl;dr:** the installed version of TensorFlow can only be used from Python.

---

I spent some time trying to get our C++ binaries compiled & linked against the installed version of TF 1.13.1. It turns out that this is impossible so I figured I may as well document the reasons why.

The TF install contains `libtensorflow_framework.so` and all the header files that we require.

I managed to get `cc:sefplay` to compile and link against these after some tweaks to our dependencies and build flags (most notably, adding explicit dependencies on protobuf to a few libraries and compiling with `--copt=-D_GLIBCXX_USE_CXX11_ABI=0`). Unfortunately, this failed at runtime because `libtensorflow_framework.so` doesn't include any `Session` factory definitions.

After some digging around on a Cloud TPU VM, I found `GrpcSession` defined in the library `_pywrap_tensorflow_internal.so` but this has a dependency on the symbol `PyExc_RuntimeError`, which is only defined in the Python executable as far as I can tell.

So that puts the kibosh on the idea of using the installed version of TensorFlow instead of compiling from source.

**Comment by tommadams** [comment]
:(
